php_phongo.c

```diff
@@ -604,36 +602,43 @@ int phongo_execute_command(mongoc_client_t *client, const char *db, const bson_t
 		return true;
 	}
 
-	/* Detect if its an command cursor */
-	if (bson_iter_init_find (&iter, doc, "cursor") && BSON_ITER_HOLDS_DOCUMENT (&iter) && bson_iter_recurse (&iter, &child)) {
-		while (bson_iter_next (&child)) {
-			if (BSON_ITER_IS_KEY (&child, "id")) {
-				cursor->rpc.reply.cursor_id = bson_iter_as_int64 (&child);
-			} else if (BSON_ITER_IS_KEY (&child, "ns")) {
+	/* This code is adapated from _mongoc_cursor_cursorid_prime(), but we avoid
+	 * advancing the cursor, since we are already positioned at the first result
+	 * after the error checking above. */
+	if (bson_iter_init_find(&iter, doc, "cursor") && BSON_ITER_HOLDS_DOCUMENT(&iter) && bson_iter_recurse(&iter, &child)) {
+		mongoc_cursor_cursorid_t *cid;
+
+		_mongoc_cursor_cursorid_init(cursor);
+		cursor->limit = 0;
+
+		cid = cursor->iface_data;
+		cid->has_cursor = true;
+
+		while (bson_iter_next(&child)) {
+			if (BSON_ITER_IS_KEY(&child, "id")) {
+				cursor->rpc.reply.cursor_id = bson_iter_as_int64(&child);
+			} else if (BSON_ITER_IS_KEY(&child, "ns")) {
 				const char *ns;
 
-				ns = bson_iter_utf8 (&child, &cursor->nslen);
-				bson_strncpy (cursor->ns, ns, sizeof cursor->ns);
-			} else if (BSON_ITER_IS_KEY (&child, "firstBatch")) {
-				if (BSON_ITER_HOLDS_ARRAY (&child)) {
-					const uint8_t *data = NULL;
-					uint32_t data_len = 0;
-					bson_t first_batch;
-
-					bson_iter_array (&child, &data_len, &data);
-					if (bson_init_static (&first_batch, data, data_len)) {
-						_mongoc_cursor_cursorid_init(cursor);
-						cursor->limit = 0;
-						cursor->is_command = false;
-						phongo_cursor_init(return_value, cursor, &first_batch, client, 1 TSRMLS_CC);
-						return true;
-					}
+				ns = bson_iter_utf8(&child, &cursor->nslen);
+				bson_strncpy(cursor->ns, ns, sizeof cursor->ns);
+			} else if (BSON_ITER_IS_KEY(&child, "firstBatch")) {
+				if (BSON_ITER_HOLDS_ARRAY(&child) && bson_iter_recurse(&child, &cid->first_batch_iter)) {
+					cid->in_first_batch = true;
 				}
```

Comment:
Ah, I forgot about [PHPC-230](https://jira.mongodb.org/browse/PHPC-230). This should certainly fix it, though, since we are now relying on the final return statement after the common `phongo_cursor_init()` below.